extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -553,6 +553,51 @@ export class Bind {
     return value;
   }
 
+  /**
+   * Returns a copy of the global state for a given field-based expression.
+   * It will wait for relevant amp-state components to finish fetching before returning.
+   *
+   * e.g. "foo.bar".
+   * @param {string} expr
+   * @return {!Promise<*>}
+   */
+  getStateWithWait(expr) {
+    const asyncLoadingAmpStates = {};
+    const ampStateEls = this.ampdoc.getRootNode().querySelectorAll('AMP-STATE');
```

Comment:
This is not the correct "root" node in all cases. See the creation of `initializePromise_`.

Reply:
Thanks for spotting this! I need to learn more about FIEs.
Done.